server/selection.go

```diff
@@ -59,6 +61,28 @@ type stakeReader interface {
 	Stakes(addrs []ethcommon.Address) (map[ethcommon.Address]*big.Int, error)
 }
 
+type dbStakeReader struct {
+	db *common.DB
```

Comment:
we can use `common.OrchestratorStore` abstraction here.

From our discussion yesterday I take away that since this is a new type using an abstraction would make writing tests for it easier.

Reply:
Fixed.